datastore/google/cloud/datastore/helpers.py

```diff
@@ -138,12 +138,12 @@ def entity_from_protobuf(pb):
         if is_list:
             exclude_values = set(value_pb.exclude_from_indexes
                                  for value_pb in value_pb.array_value.values)
-            if len(exclude_values) != 1:
+            if len(exclude_values) > 1:
```

Comment:
I keep playing with this from time to time but I'm having a hard time testing it.
You can't load an empty value into the generated proto objects so I need to figure out how to mock the response from `Connection` in order to replicate the situation. I haven't dug into the protos to figure out what the response should look like yet.